lib/services/local/users.go

```diff
@@ -1470,17 +1460,27 @@ func (s *IdentityService) GetKeyAttestationData(ctx context.Context, publicKey c
 		return nil, trace.BadParameter("missing parameter publicKey")
 	}
 
-	sshPub, err := ssh.NewPublicKey(publicKey)
+	pubDER, err := x509.MarshalPKIXPublicKey(publicKey)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	key := ssh.FingerprintSHA256(sshPub)
+	key := fingerprintSha256(pubDER)
 	item, err := s.Get(ctx, backend.Key(attestationsPrefix, key))
-	if err != nil {
+	if trace.IsNotFound(err) {
+		// Fallback to old fingerprint (std base64 encoded ssh public key) for backwards compatibility.
+		// DELETE IN 13.0.0
+		key, err = oldFingerprintSha256(publicKey)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		item, err = s.Get(ctx, backend.Key(attestationsPrefix, key))
 		if trace.IsNotFound(err) {
 			return nil, trace.NotFound("hardware key attestation not found")
+		} else if err != nil {
+			return nil, trace.Wrap(err)
 		}
+	} else if err != nil {
 		return nil, trace.Wrap(err)
 	}
```

Comment:
Suggestion: Simplify a bit?

```suggestion
		}
	}
	if err != nil {
		return nil, trace.Wrap(err)
	}
```